Sources/Hypertext.swift

```diff
@@ -6,18 +6,23 @@
 //  Copyright © 2016 Sahand Nayebaziz. All rights reserved.
 //
 
-public protocol Renderable {
-    func render() -> String
-    func render(startingWithSpacesCount: Int) -> String
+public protocol Renderable: CustomStringConvertible {
+    func render(indent: Int?, startingWithSpacesCount: Int) -> String
 }
 
-extension CustomStringConvertible {
-    public func render() -> String {
-        return String(describing: self)
+public extension Renderable {
+    var description: String {
+        return render()
+    }
+
+    func render(indent: Int? = nil) -> String {
+        return render(indent: indent, startingWithSpacesCount: 0)
```

Comment:
This is starting to make more sense to me. Can you explain this so I can understand?